### PR TITLE
Enumerable

### PR DESCRIPTION
**Pull request by dannkkie**
Finished enumerable project please kindly review for further action

### CODE REVIEW

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  # this is the my_each_with_index method
+  def my_each_with_index
```
Comment (Forison) [inline enumerable.rb]: * [ ] must return enumerator when no block is given
[Check docs for functionalies](https://ruby-doc.org/core-2.7.0/Enumerable.html#method-i-each_with_index)
```  p [1,2,3,4,5,6].my_each_with_index  # => #<Enumerator: [1, 2, 3, 4, 5]:my_each_with_index>```

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  # this is the my_select method
+  def my_select
```
Comment (Forison) [inline enumerable.rb]: * [ ] must return enumerator when no block is given
[Check docs for functionalies](https://ruby-doc.org/core-2.7.0/Enumerable.html#method-i-each_with_index)
```  p [1,2,3,4,5,6].my_select  # => #<Enumerator: [1, 2, 3, 4, 5]:my_select>```

File: enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    selected = []
+    my_each { |i| selected << i if yield(i) }
+    selected
+  end
+
+  # this is the my_all method
+  def my_all?
```
Comment (Forison) [inline enumerable.rb]: * [ ] my_all? is not returning the right answers for the exmples below
[Check docs for functionalies](https://ruby-doc.org/core-2.7.0/Enumerable.html#method-i-each_with_index)
```
p %w[ant bear cat].my_all?(/t/)                        #=> false
p [1, 2i, 3.14].my_all?(Numeric)                      #=> true
p [nil, true, 99].my_all?                                   #=> false
 
```

File: enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    boolean = true
+    my_each do |i|
+      boolean = false unless yield(i)
+      break if boolean == false
+    end
+    boolean
+  end
+
+  # this is the my_any method
+  def my_any?
```
Comment (Forison) [inline enumerable.rb]: * [ ] my_any? is not returning the right answers for the exmples below
[Check docs for functionalies](https://ruby-doc.org/core-2.7.0/Enumerable.html#method-i-each_with_index)
```
p %w[ant bear cat].my_any?(/d/)                        #=> false
p [nil, true, 99].my_any?(Integer)                     #=> true
p [nil, true, 99].my_any?                              #=> true

```

File: enumerable.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    boolean = false
+    my_each { |i| boolean = true if yield(i) }
+    boolean
+  end
+
+  # this is the my_none method
+  def my_none?
```
Comment (Forison) [inline enumerable.rb]: * [ ] my_none? is not returning the right answers for the exmples below
[Check docs for functionalies](https://ruby-doc.org/core-2.7.0/Enumerable.html#method-i-each_with_index)
```
p %w{ant bear cat}.my_none?(/d/)                        #=> true
p [1, 3.14, 42].my_none?(Float)                         #=> false
p [nil].my_none?                                        #=> true
p [nil, false].my_none?                                 #=> true
p [nil, false, true].my_none?                           #=> false

```

File: enumerable.rb [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    boolean = true
+    my_each do |i|
+      boolean = false if yield(i)
+      break if boolean == false
+    end
+    boolean
+  end
+
+  # this is the my_count method
+  def my_count
```
Comment (Forison) [inline enumerable.rb]: * [ ] my_count is not returning the right answers for the example below
[Check docs for functionalies](https://ruby-doc.org/core-2.7.0/Enumerable.html#method-i-each_with_index)
```
p ary.my_count(2)            #=> 2
```

File: enumerable.rb [hunk 7]
```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module Enumerable
+  # this is the my_each method
+  def my_each
```
[the remark this answers is not in the document]
Comment (dannkkie) [reply enumerable.rb]: done

File: enumerable.rb (hunk 1 above)
Comment (dannkkie) [reply enumerable.rb]: done

File: enumerable.rb (hunk 2 above)
Comment (dannkkie) [reply enumerable.rb]: done

File: enumerable.rb (hunk 3 above)
Comment (dannkkie) [reply enumerable.rb]: done

File: enumerable.rb (hunk 4 above)
Comment (dannkkie) [reply enumerable.rb]: done

File: enumerable.rb (hunk 5 above)
Comment (dannkkie) [reply enumerable.rb]: done

File: enumerable.rb (hunk 6 above)
Comment (dannkkie) [reply enumerable.rb]: done

**Comment by dannkkie** [comment]
Please kindly review all changes made

**Comment by dannkkie** [comment]
if you look at my repo i have master and the feature branch
Master branch: https://github.com/dannkkie/enumerable
Feature branch: https://github.com/dannkkie/enumerable/tree/enumerable
I followed the git work flow please

**Pull request closed without merging**